objects_diff.go

```diff
@@ -0,0 +1,108 @@
+package git
+
+import (
+	"io"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ObjectsDiff provides a way to compare two Commit arrays and gets the differences
+// (new commits, trees or files)
+type ObjectsDiff struct {
+	repository *Repository
+}
+
+// NewObjects returns hashes that are into the serverHas commits histories
+// and not into have commits histories
+func (od *ObjectsDiff) NewObjects(have []*Commit, remoteHas []*Commit) ([]plumbing.Hash, error) {
```

Comment:
Usually in git, what you have is call `our` and the "remote" storage is call `their`